x-pack/test_serverless/api_integration/test_suites/common/saved_objects_management/relationships.ts

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { schema } from '@kbn/config-schema';
+import { SavedObjectRelation } from '@kbn/saved-objects-management-plugin/public';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { RoleCredentials } from '../../../../shared/services';
+
+// TODO: convert to supertestWithoutAuth
```

Comment:
I'm not sure which approach to take.

Reply:
I'm not sure I understand the alternatives here. Would you mind explaining the decision you're considering?

Reply:
`supertest` has embedded credentials, whereas `supertestWithoutAuth` doesn't. One can use `supertestWithoutAuth` to create users with a [specific role](https://github.com/elastic/kibana/blob/825efd0ef3e8940a8d40e2f9c99986a5cd1276a3/x-pack/test_serverless/api_integration/test_suites/common/platform_security/request_as_viewer.ts#L20):

```
// get auth header for Viewer role
credentials = await svlUserManager.getApiCredentialsForRole('viewer');
await supertestWithoutAuth
        .get('/internal/security/me')
        .set(svlCommonApi.getInternalRequestHeader())
        .set(credentials);
```

Reply:
fixed in [`e10b03e`](https://github.com/elastic/kibana/pull/184888/commits/e10b03eb15678edf35b3da8afb1b3a4a5d467aac)